harvesters/src/main/java/org/fao/geonet/component/harvester/csw/Harvest.java

```diff
@@ -831,6 +832,9 @@ public static Protocol validate(String string) {
             this.harvester = harvester;
             this.responseHandler = responseHandler;
             this.serviceContext = serviceContext;
+            if(!(serviceContext instanceof ServiceContext.AppHandlerServiceContext)){
+                serviceContext.warning("Harvest required to work with shared \"harvester\" service context, '"+serviceContext.getService()+"' provided.");
```

Comment:
This is what I indicated as a question for testing above.

I am not sure *if* this can happen, the warning is here in case it does. And also so we can put a breakpoint here to catch it when it does.

We could this a solid failure if we wish to be struck about this.